gateway/gateway_test.go

```diff
@@ -763,7 +763,7 @@ func TestErrorBubblingFromBackend(t *testing.T) {
 		})
 	}
 
-	testError("404 Not Found from IPLD", &ipld.ErrNotFound{}, http.StatusNotFound)
+	testError("500 Not Found from IPLD", &ipld.ErrNotFound{}, http.StatusInternalServerError)
```

Comment:
I think this was a mistake. `ipld.ErrNotFound` is an "I can't find this block"  kind of error which IIUC is supposed to be 5xx as opposed to "this path doesn't exist/IPLD errors" which are 4xx.

I think this error could also reasonably be a `502`